# Exercise 01: Design a detector for banana cargo at port.

A- Calculate the neutrino flux from a ship of Bananas. On average, a ship carrying bananas might spend 3 days in port just for offloading.

B- Suggest possible reaction channels to detect the flux?

C- Discuses why the inverse beta decay is not a possible solution.

#### Your answer:

A. For calculating the flux, we first have to calculate the fraction-decayed or decay rate of Potassium-40 in bananas
and then calculate the flux at an approximate distance of the cargo from the detector.
The cargo might spend 3 days in port, but we can show that the amount of time is irrelevant,
because the half-life of this particular cargo is very long.  

The decay rate formula is:  

$$
A(t) = \lambda N(t) = \lambda N_0 e^{-\lambda t},
$$

and $\lambda$ is the decay constant, which is:
$$
\lambda = \frac{\ln 2}{T_{1/2}}
$$
Since the lifetime of Potassium-40 is much longer than the experimental time (3 days),
the $\lambda t$ factor is much less than 1, and the activity can be approximated as $\lambda N$.  

The given numbers are found in Wikipedia and other reliable websites.  

Potassium-40 has two dominant decay channels:  

1. Electron emission from beta-minus decay (89%), which produces a $\bar{ν}_{e}$.  
2. Electron capture (10%), which produces a $ν_{e}$.  

We can separate the flux of these two channels using their decay fractions,
but here I have only calculated the total flux.


In [1]:
import numpy as np

# Given
banana_mass = 1000000 # Total Mass of the Cargo (grams)
Kfrac = 0.004 #  Amount of Potassuim in 1 gram banana
K40frac = 0.000117

mass_k40 = banana_mass * Kfrac * K40frac # grams

molar_mass_k40 = 40  # grams per mole
avogadro_number = 6.022e23  # atoms per mole

# Calculate number of atoms
number_atoms = (mass_k40 / molar_mass_k40) * avogadro_number
print(f"Total K-40 mass: {mass_k40:.6f} g")
print(f"Number of K-40 atoms: {number_atoms:.3e}")



Total K-40 mass: 0.468000 g
Number of K-40 atoms: 7.046e+21


In [2]:
# Given
half_life_years = 1.25e9  # years
seconds_per_year = 3.154e7  # seconds in one year

half_life_sec = half_life_years * seconds_per_year

# Calculate decay constant
decay_constant = np.log(2) / half_life_sec
print(f"Decay constant (1/s): {decay_constant:.3e}")

Decay constant (1/s): 1.758e-17


In [4]:
time_days = 3
seconds_per_day = 86400

time_sec = time_days * seconds_per_day

# Calculate Avrage Activity

fraction_decayed= 1 - np.exp(-decay_constant * time_sec)
number_decays = number_atoms * fraction_decayed
Activity_avg = number_decays / time_sec

# Or simply just ...

Total_Activity = decay_constant * number_atoms

print(f"Fraction decayed in 3 days: {fraction_decayed:.3e}")
print(f"Total decays in 3 days: {number_decays:.3e}")
#--------Activity----------
#Both should be the same#
print(f"Avrage Activity in 3 days : {Activity_avg:.3e}")
print(f"Total Activity : {Total_Activity:.3e}")



Fraction decayed in 3 days: 4.557e-12
Total decays in 3 days: 3.211e+10
Avrage Activity in 3 days : 1.239e+05
Total Activity : 1.239e+05


In [5]:
# Distance from source in meters
r = 10 # Very Hypothetical Number

# Calculate flux
flux = Total_Activity / (4 * np.pi * r**2)


# neutrinos per square meter per second
print(f"Neutrino flux at 10 m: {flux :.3f} neutrinos/m²/s")



Neutrino flux at 10 m: 98.576 neutrinos/m²/s


B. I only focus on the two dominant decay channels of Potassium-40 mentioned above,
because other channels have a very low contribution.  


### 1. Beta-minus decay

$$
^{40}\text{K} \to {}^{40}\text{Ca} + e^- + \bar{\nu}_e
$$  

It produces a $\bar{\nu}_{e}$ with energies up to 1.3 MeV.  

**Possible detection channels:**

- **Inverse beta decay (IBD):**  

$$
\bar{\nu}_e + p \to e^+ + n
$$

I explain why this isn’t feasible in Section C of the question.  

- **Neutrino-electron elastic scattering:**  

$$
\bar{\nu} + e^- \to \bar{\nu} + e^-
$$  

It works for low-energy neutrinos but has a very small cross section.



### 2. Electron capture decay

$$
^{40}\text{K} + e^- \to {}^{40}\text{Ar} + \nu_e
$$  

It produces $\nu_{e}$ with energy ~1.5 MeV.  

**Possible detection channels:**

- **Neutrino-electron elastic scattering:**  

$$
\nu_e + e^- \to \nu_e + e^-
$$
Same problem as above.  

- **Charged-current reactions on nuclei:**  

Low-energy $\nu_e$ can interact with certain nuclei, for example:  

$$
\nu_e + {}^{37}\text{Cl} \to {}^{37}\text{Ar} + e^-
$$  

This is the principle used in solar neutrino detectors (Cl or Ga detectors).  


Bottom line is conventional neutrino detectors like water Cherenkov detectors are mostly **insensitive** to these low-energy neutrinos.  

Moreover, the **background at a port** from the Sun, people, and other sources is extremely high and creating a very isolated environment for detection would be inefficient.  

Probably this is why we **do not have a "banana neutrino detector"**. However, from my research, detecting bananas using **image processing and robotics** for harvesting is an active area of development. Most of the work I found on this topic involves **computer vision techniques** rather than any physics-based detection.  

It is worth mentioning that, if we put aside neutrinos, we **can detect a flux of 1.46 MeV gamma rays** from the electron capture decay channel. In this process, the nucleus absorbs an inner orbital electron and leaves the daughter nucleus ${}^{40}\text{Ar}$ in an **excited state**. The excited nucleus then relaxes to the ground state by emitting a **gamma photon of 1.46 MeV**. Detection of these gammas is much easier than neutrinos, but the count rate is still very low.




C. Inverse beta decay (IBD) requires the incoming antineutrino to have an energy greater than **1.8 MeV** to produce a positron and a neutron:  

$$
\bar{\nu}_e + p \to e^+ + n
$$  

However, the electron antineutrinos emitted from $^{40}\text{K}$ beta-minus decay have **maximum energies of only ~1.31 MeV**, which is below the IBD threshold. Therefore, it is an impossible detection method for banana neutrinos.



---

# Exercise 02: Bayesian Parameter Estimation via MCMC

**Context:**  
In observational cosmology, the relationship between redshift $z$ and luminosity distance $D_L$ in a flat $\Lambda$CDM model is approximately:

$$ D_L(z; H_0, \Omega_m) \approx \frac{c(1+z)}{H_0} \int_0^z \frac{dz'}{\sqrt{\Omega_m(1+z')^3 + (1-\Omega_m)}} $$

where $c = 299{,}792.458\ \text{km/s}$.

In this exercise, you will be given synthetic $D_L$ vs. $z$ data generated for realistic cosmological parameters.  
Your task is to use **Metropolis–Hastings MCMC** to estimate $H_0$ and $\Omega_m$ from this noisy dataset.

---

## What to Do

### 1. Model and Likelihood
- Assume Gaussian errors $\sigma_{D_L}$ for each data point.
- The likelihood for parameters $(H_0, \Omega_m)$ is:

$$ L(H_0, \Omega_m) \propto \exp\left( -\frac{1}{2} \sum_{i} \frac{\left[ D_{L,i}^{\text{obs}} - D_L(z_i; H_0, \Omega_m) \right]^2}{\sigma_{D_L}^2} \right) $$

---

### 2. Set Prior Distributions as:
- $H_0 \sim \mathcal{U}(50,\, 90)$ km/s/Mpc  
- $\Omega_m \sim \mathcal{U}(0.1,\, 0.6)$

---

### 3. Metropolis–Hastings Implementation
- Proposal: Gaussian jumps in both parameters:

$$ H_0^{\text{new}} \sim \mathcal{N}(H_0^{\text{old}}, \delta_{H_0}^2), \quad \Omega_m^{\text{new}} \sim \mathcal{N}(\Omega_m^{\text{old}}, \delta_{\Omega_m}^2) $$

- Compute acceptance ratio:

$$ r = \frac{P_{\text{posterior}}\left(H_0^{\text{new}}, \Omega_m^{\text{new}}\right)}{P_{\text{posterior}}\left(H_0^{\text{old}}, \Omega_m^{\text{old}}\right)} $$

- Accept or reject according to $u < r$, where $u \sim U(0,1)$.

---

### 4. Tasks
- Write your own Metropolis–Hastings routine from scratch
- Run chains long enough to achieve convergence.
- Plot posterior distributions and credible intervals for $H_0$ and $\Omega_m$.
- Discuss how changing proposal widths $\delta_{H_0}$ and $\delta_{\Omega_m}$ affects convergence.

---

### Data Format
You will be provided with a TXT file **`mcmc_cosmo_data.txt`** located in the `data` folder.
Load the file (ignoring the header) and use it in your MCMC implementation.


#### Your answer: